src/Microsoft.ComponentDetection.Detectors/spdx/Spdx22ComponentDetector.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.Internal;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.ComponentDetection.Detectors.Spdx
+{
+    /// <summary>
+    /// Spdx22ComponentDetector discover SPDX SBOM files and create components with the information about
+    /// what SPDX document describes. 
+    /// </summary>
+    [Export(typeof(IComponentDetector))]
+    public class Spdx22ComponentDetector : FileComponentDetector, IDefaultOffComponentDetector 
+    {
+        public override string Id => "SPDX22SBOM";
+        
+        public override IEnumerable<string> Categories =>
+            new[] { Enum.GetName(typeof(DetectorClass), DetectorClass.Spdx) };
+        
+        public override IEnumerable<ComponentType> SupportedComponentTypes { get; } = new[] { ComponentType.Spdx };
+
+        public override int Version => 1;
+        
+        public override IList<string> SearchPatterns { get; } = new List<string> { "*.spdx.json" };
+
+        private readonly IEnumerable<string> supportedSPDXVersions = new List<string> { "SPDX-2.2" };
+
+        protected override Task OnFileFound(ProcessRequest processRequest, IDictionary<string, string> detectorArgs)
+        {
+            Logger.LogVerbose($"Discovered SPDX2.2 manifest file at: {processRequest.ComponentStream.Location}");
+            var file = processRequest.ComponentStream;
+
+            try
+            {
+                var hash = GetSHA1HashFromStream(file.Stream);
+
+                // Reset buffer to starting position after hash generation.
+                file.Stream.Seek(0, SeekOrigin.Begin);
+
+                using (StreamReader sr = new StreamReader(file.Stream))
+                using (JsonReader reader = new JsonTextReader(sr))
+                {
+                    JsonSerializer serializer = new JsonSerializer();
+
+                    try
+                    {
+                        JObject document = (JObject)serializer.Deserialize(reader);
```

Comment:
nit: there's a generic version of this method available
```suggestion
                        var document = serializer.Deserialize<JObject>(reader);
```